app/questionnaire/questionnaire_store_updater.py

```diff
@@ -209,38 +222,125 @@ def update_section_status(
         )
         self._progress_store.update_section_status(status, section_id, list_item_id)
 
-    def started_section_keys(self, section_ids: Iterable[str] = None):
+    def started_section_keys(
+        self, section_ids: Iterable[str] = None
+    ) -> list[SectionKeyType]:
         return self._progress_store.section_keys(
             statuses={CompletionStatus.COMPLETED, CompletionStatus.IN_PROGRESS},
             section_ids=section_ids,
         )
 
+    def _update_answer(
+        self,
+        answer_id: str,
+        list_item_id: Optional[str],
+        answer_value: AnswerValueTypes,
+    ):
+        answer_value_to_store = (
+            {
+                key: value
+                for key, value in answer_value.items()
+                if value not in self.EMPTY_ANSWER_VALUES
+            }
+            if isinstance(answer_value, dict)
+            else answer_value
+        )
+
+        if answer_value_to_store in self.EMPTY_ANSWER_VALUES:
+            answer_updated = self._answer_store.remove_answer(
+                answer_id, list_item_id=list_item_id
+            )
+        else:
+            answer = Answer(
+                answer_id=answer_id,
+                list_item_id=list_item_id,
+                value=answer_value_to_store,
+            )
+
+            answer_updated = self._answer_store.add_or_update(answer)
+
+        return answer_updated
+
+    def _capture_dependencies_for_answer(self, answer_id):
```

Comment:
True, but we did say we would try to add them for new functions so the burden isn't as great when we did get around to them

Reply:
Ya, I agree, but not if the file is not enforced. i.e for new files it would make sense. For example, I could say this function returns a `dict` when it doesn't and mypy won't error. That's why I was suggesting for me to just add all the missing type hints, only about 19 left. Happy for me to add it?